src/python/pants/backend/python/goals/pytest_runner.py

```diff
@@ -509,7 +510,11 @@ async def run_python_tests(
     setup = await Get(
         TestSetup, TestSetupRequest(batch.elements, batch.partition_metadata, is_debug=False)
     )
-    result = await Get(FallibleProcessResult, Process, setup.process)
+    HARDCODED_RETRY_COUNT = 5  # TODO: get from global option or batch
```

Comment:
+1

And possibly provide it via the `PyTestRequest.Batch` if we can imagine a case where some batches would be using different numbers of retries?

I expect that retries should be disabled by default, but that we recommend enabling them in CI on https://www.pantsbuild.org/docs/using-pants-in-ci

Reply:
I agree, it would be great to be able to flag individual test targets as extra-flaky and give them multiple retries. Similar to how timeouts are currently done. Could we defer that to a future MR, maybe at the same time as we add retries to the other test types?

Reply:
If we land a hardcoded retry like this, we would have to fast follow tomorrow (or at least this week...?) to add the option. If you can commit to that, then yea.

Reply:
now it pulls it from the global option `attempts_default`. I think this thread is tracking an older commit